connect_ext_ppr/webapp.py

```diff
@@ -465,6 +482,74 @@ def list_hubs_by_product(
             reponse_list.append(HubSchema(id=hub['id'], name=hub['name']))
         return reponse_list
 
+    @router.get(
+        '/deployments/{deployment_id}/pricing/batches',
+        summary='List Pricing Batch for Deployment',
+        response_model=List[BatchSchema],
+    )
+    def get_deployment_batches(
+        self,
+        deployment_id: str,
+        db: VerboseBaseSession = Depends(get_db),
+        client: ConnectClient = Depends(get_installation_client),
+        installation: dict = Depends(get_installation),
+    ):
+        deployment = get_deployment_by_id(deployment_id, db, installation)
+        marketplace_ids = identify_marketplaces(
+            client,
+            deployment.hub_id,
+        )
+
+        batches = list(client('pricing').batches.filter(
+            R().stream.owner.id.eq(deployment.account_id),
+            R().stream.context.product.id.eq(deployment.product_id),
+            R().stream.context.marketplace.id.in_(marketplace_ids),
+            R().test.ne(True),
+            R().status.eq('published'),
+        ))
+
+        return [BatchSchema(**b) for b in batches]
+
+    @router.post(
+        '/deployments/{deployment_id}/pricing/batches/{batch_id}/process',
+        summary='Process Pricing Batch for Deployment',
+        response_model=BatchProcessResponseSchema,
+    )
+    def process_pricing_batch(
+        self,
+        deployment_id: str,
+        batch_id: str,
+        client: ConnectClient = Depends(get_installation_client),
+        db: VerboseBaseSession = Depends(get_db),
+        cbc_db: Session = Depends(get_cbc_extension_db),
+        installation: dict = Depends(get_installation),
+        logger: LoggerAdapter = Depends(get_logger),
+    ):
+        try:
+            deployment = get_deployment_by_id(deployment_id, db, installation)
+            batch = fetch_and_validate_batch(client, batch_id, deployment)
+            reseller_id = identify_reseller_id(client, batch, deployment)
+            file_name, dataset = prepare_file(client, batch_id)
+
+            data_id = process_batch(
+                cbc_db,
+                file_name,
+                reseller_id,
+                deployment,
+                dataset,
+            )
+
+            response = BatchProcessResponseSchema(
+                task_info=f'/flat-catalog/price-import-wizard/{data_id}/set-prices',
+            )
+            return JSONResponse(
+                status_code=202,
+                content=response.dict(),
+            )
+        except ClientError as e:
+            logger.exception(f'Error while uploading price file for {batch_id}')
+            return JSONResponse(status_code=400, content=e.json if e.json else {})
```

Comment:
[COD-C] Leo, here there is no need to return the `ClientError` wrapped in a `JSONResponse` since eaas runner web handler is already doing it for you when setup the `exception_handlers` of the app [here](https://github.com/cloudblue/connect-extension-runner/blob/master/connect/eaas/runner/handlers/web.py#L162C22-L162C22) if the error is instance of ClientError.

Reply:
This ClientError is from CBC Client, that is different from ClientError of connect open api client. So, I do not think any change is needed.